custom-plugins/cms131v14_diabetes_eye_exam/protocols/cms131v14_protocol.py

```diff
@@ -0,0 +1,690 @@
+
+    def _is_diabetes_code(self, coding) -> bool:
+        try:
+            normalized_system = coding.system.replace("-", "").upper()
+            normalized_code = coding.code.replace(".", "")
+
+            if normalized_system == "ICD10" or normalized_system == "ICD10CM":
+                diabetes_codes = getattr(Diabetes, "ICD10CM", set())
+                diabetes_codes_normalized = {code.replace(".", "") for code in diabetes_codes}
+                return normalized_code in diabetes_codes_normalized
+            elif normalized_system == "SNOMEDCT" or normalized_system == "SNOMED":
+                diabetes_codes = getattr(Diabetes, "SNOMEDCT", set())
+                return coding.code in diabetes_codes
+
+            return False
+
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking if code is diabetes: {str(e)}")
+            return False
+
+    def _is_condition_diabetes(self, condition: Condition) -> bool:
+        try:
+            condition_codings = ConditionCoding.objects.filter(condition=condition)
+            return any(self._is_diabetes_code(coding) for coding in condition_codings)
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking if condition is diabetes: {str(e)}")
+            return False
+
+    def _should_remove_card(self, patient: Patient) -> bool:
+        try:
+            condition_id = self.event.target.id
+            condition = Condition.objects.filter(id=condition_id).first()
+
+            if not condition:
+                log.warning(f"CMS131v14: Could not find condition {condition_id}")
+                return False
+
+            if not self._is_condition_diabetes(condition):
+                log.info(f"CMS131v14: Condition {condition_id} is not diabetes, no action needed")
+                return False
+
+            if condition.entered_in_error:
+                log.info(
+                    f"CMS131v14: Diabetes condition {condition_id} marked as entered_in_error for patient {patient.id}"
+                )
+                if not self._has_diabetes_diagnosis(patient):
+                    log.info(f"CMS131v14: No other diabetes diagnoses found, removing card")
+                    return True
+                else:
+                    log.info(f"CMS131v14: Patient still has other diabetes diagnoses, keeping card")
+                    return False
+
+            return False
+
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking if card should be removed: {str(e)}")
+            return False
+
+    def compute(self) -> list[Effect]:
+        try:
+            patient_id = self._get_patient_id()
+            if not patient_id:
+                log.warning("CMS131v14: Could not determine patient ID from event, skipping")
+                return []
+
+            log.info(f"CMS131v14: Processing event for patient_id={patient_id}")
+
+            patient = Patient.objects.filter(id=patient_id).first()
```

Comment:
`_get_patient_id()` looks up the patient, returns only the id, then we use that value to look up the patient again. If the method was instead `_get_patient()` we could save a duplicate query.